File: 002 Add Two Numbers.java
/** 2	Add Two Numbers
 * You are given two linked lists representing two non-negative numbers. 
 * The digits are stored in reverse order and each of their nodes contain a single digit. 
 * Add the two numbers and return it as a linked list.
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode dummyHead = new ListNode(0);
        ListNode p1 = l1, p2 = l2, prevNode = dummyHead;
        int carry = 0;
        while(p1 != null || p2 != null){
            int v1 = (p1 == null) ? 0 : p1.val,
                v2 = (p2 == null) ? 0 : p2.val,
                curr = v1 + v2 + carry;
            carry = curr / 10;
            prevNode.next = new ListNode(curr % 10);
            prevNode = prevNode.next;
            if (p1 != null) p1 = p1.next;
            if (p2 != null) p2 = p2.next;
        }
        if(carry > 0){
            prevNode.next = new ListNode(carry);
        }
        return dummyHead.next;
    }
}

File: JavaScript/002 Add Two Numbers.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
    var dummyHead = new ListNode(0);
    var tail = dummyHead;
    var carry = 0;
    while(l1 !== null || l2 !== null){
        if(l1 !== null){carry += l1.val; l1 = l1.next;}
        if(l2 !== null){carry += l2.val; l2 = l2.next;}
        tail.next = new ListNode(carry % 10);
        tail = tail.next;
        carry = (carry > 9 ? 1 : 0);
    }
    if(carry) tail.next = new ListNode(carry);
    return dummyHead.next;
    
};